base/rtlib.jl

```diff
@@ -0,0 +1,194 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+"""
+    RTLIB
+
+Implements the runtime library for Julia. The implementations are based on
+llvm's compiler-rt. This implementations follows the compiler-rt naming convention
+and registers the pure Julia implementation as `extern_c` so that LLVM can find them.
+
+As a secondary interface `RTLIB.convert(::Type{T}, x)` is provided.
+"""
+module RTLIB
+
+register(f::Function, rtype::ANY, argt::ANY, name::String) =
+    ccall(:jl_extern_c, Void, (Any, Any, Any, Cstring),
+          f, rtype, argt, name)
+
+include("rtlib/fp_util.jl")
```

Comment:
I think we're moving towards having submodules be in their own folders under base, including their top-level files - so `rtlib/RTLIB.jl`